### PR TITLE
Uncommitted confirmation

### LINKED ISSUE #1669: Uncommitted changes warning only appears on the model switcher
When you switch models from the model switcher it gives a warning if you have uncommitted changes. This warning does not appear if you switch models from the profile etc.

Possibly we should move the warning to a confirmation popup.

### PR DESCRIPTION
**Pull request by huwshimi**
Show a confirmation popup when switching between models when there are uncommitted changes. Fixes #1669. Also required by #1652.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- add a service or machine
- try and switch models via the model switcher
- you should get a confirmation popup
- add a service or machine
- try and switch models via the profile
- you should get a confirmation popup

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5087/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5088/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5089/
Test PASSed.

**Comment by hatched** [comment]
When you're in the profile view I don't think there is a way to get back to the canvas in that model without first closing and creating a new connection which effectively makes the ECS invalid because it'll point to different objects in the db. Did you not find this?

AIUI once you enter the profile view you're essentially disconnecting from the active environment. So maybe the warning should be on opening the profile?

File: jujugui/static/gui/src/app/views/utils.js [hunk 1]
```diff
@@ -1924,6 +1924,74 @@ YUI.add('juju-view-utils', function(Y) {
   */
   utils.switchModel = function(
     createSocketURL, switchEnv, env, uuid, modelList, name, callback) {
+    var switchModel = utils._switchModel.bind(this,
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/utils.js]: Nice!

**Comment by hatched** [comment]
:+1: Great cleanup!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5106/
Test PASSed.

**Comment by kadams54** [comment]
Great solution! :+1: on the code.

**Comment by huwshimi** [comment]
Thanks @hatched and @kadams54. I'm going to do a followup to disconnect the model when opening the profile. :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**